server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/OdcJobListener.java

```diff
@@ -165,14 +165,18 @@ public void jobWasExecuted(JobExecutionContext context, JobExecutionException jo
         ScheduleTaskContextHolder.clear();
         Optional<ScheduleEntity> scheduleEntityOptional =
                 scheduleRepository.findById(ScheduleTaskUtils.getScheduleId(context));
-        if (scheduleEntityOptional.isPresent()) {
-            ScheduleEntity scheduleEntity = scheduleEntityOptional.get();
-            if (jobException != null && notificationProperties.isEnabled()) {
-                try {
+        if (notificationProperties.isEnabled() && scheduleEntityOptional.isPresent()) {
+            try {
+                ScheduleEntity scheduleEntity = scheduleEntityOptional.get();
+                if (jobException != null) {
                     broker.enqueueEvent(eventBuilder.ofFailedTask(scheduleEntity));
-                } catch (Exception e) {
-                    log.warn("Failed to enqueue event.", e);
+                } else if (!taskFrameworkEnabledProperties.isEnabled()
```

Comment:
Why should the notification module need to be aware of the task framework?

Reply:
This is not in notification module. It's in schedule module, to determine whether to call the notification module.
The notification module is still not aware of task framework.